server/src/main/java/org/opensearch/search/query/ConcurrentQueryPhaseSearcher.java

```diff
@@ -80,12 +80,26 @@ private static boolean searchWithCollectorManager(
         try {
             final ReduceableSearchResult result = searcher.search(query, collectorManager);
             result.reduce(queryResult);
-        } catch (EarlyTerminatingCollector.EarlyTerminationException e) {
-            queryResult.terminatedEarly(true);
-        } catch (TimeExceededException e) {
+        } catch (RuntimeException re) {
+            // Rethrow exception if cause is null
+            if (re.getCause() == null) {
+                throw re;
+            }
+
+            // Unwrap the RuntimeException and ExecutionException from Lucene concurrent search method and rethrow
+            if (re.getCause() instanceof ExecutionException || re.getCause() instanceof InterruptedException) {
```

Comment:
We could incorporate this logic into `rethrowCauseIfPossible(re)`

Reply:
That's a good idea, I think it makes sense to move as much of the exception handling as possible into the rethrow method, I will make this change.